Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM debian:jessie
+
+ENV ZCASH_CONF=/home/zcash/.zcash/zcash.conf
+ENV ZCASH_DATA /home/zcash/.zcash
+
+RUN set -e && \
+  apt-get update -q && \
+  apt-get install -q -y apt-transport-https wget && \
+  wget -qO - https://apt.z.cash/zcash.asc | apt-key add - && \
+  echo "deb [arch=amd64] https://apt.z.cash/ jessie main" | tee /etc/apt/sources.list.d/zcash.list && \
+  apt-get update -q && \
+  apt-get install zcash -q -y && \
+  zcash-fetch-params && \
+  groupadd -r zcash && \
+  useradd -r -m -g  zcash zcash && \
+  mv /root/.zcash-params /home/zcash/ && \
+  mkdir -p /home/zcash/.zcash/ && \
+  chown -R zcash:zcash /home/zcash && \
+  apt-get purge -q -y apt-transport-https wget && \
+  apt-get clean -q -y && \
+  apt-get autoclean -q -y && \
+  apt-get autoremove -q -y && \
+  rm -rf "/var/lib/apt/lists/*" "/var/lib/apt/lists/partial/*" "/tmp/*" "/var/tmp/*"
+
+VOLUME /home/zcash/.zcash
+
+COPY docker-entrypoint.sh /entrypoint.sh
+RUN chmod +x /entrypoint.sh
+ENTRYPOINT ["/entrypoint.sh"]
+
+USER zcash
```

Comment:
You probably want to delay this to the docker-entrypoint with gosu, so you can run `chown -R zcash:zcash /home/zcash` again in the entrypoint to ensure possibly root owned volumes get corrected. :)

Reply:
Sorry for the late response, but i am not sure if i understand this correctly.
You mean if someone already has a data volume which is owned by root this image should change the permissions of the data to zcash:zcash?

Reply:
If you execute `docker -v /somewhere:/home/zcash/.zcash yourimage` and /somewhere doesn't exist yet, docker is going to create this folder for you, but zcash is not able to write to it because the folder is owned by root. After you chowned the folder, the permissions stay correct.